app/src/main/java/com/sunasterisk/travelapp/di/Injector.kt

```diff
@@ -0,0 +1,68 @@
+import com.sunasterisk.travelapp.data.source.local.dao.ReservationDAOImpl
+import com.sunasterisk.travelapp.data.source.local.dao.UserDAOImpl
+import com.sunasterisk.travelapp.data.source.local.database.DatabaseHelper
+import com.sunasterisk.travelapp.data.source.local.preferences.PreferencesHelperImpl
+import com.sunasterisk.travelapp.data.source.remote.HotelRemoteDatasource
+import com.sunasterisk.travelapp.data.source.remote.LocationRemoteDatasource
+import com.sunasterisk.travelapp.data.source.remote.RestaurantRemoteDatasource
+import com.sunasterisk.travelapp.utils.Database.DB_NAME
+import com.sunasterisk.travelapp.utils.Database.DB_VERSION
+
+object Injector {
+    fun getBookingRepository(context: Context): BookingRepository {
+        val database = DatabaseHelper.getInstance(DB_NAME, DB_VERSION, context)
+        val bookingDAO = BookingDAOImpl.getInstance(database)
+        val local = BookingLocalDatasource.getInstance(bookingDAO)
+        return BookingRepositoryImpl.getInstance(local)
+    }
+
+    fun getReservationRepository(context: Context): ReservationRepository {
+        val database = DatabaseHelper.getInstance(DB_NAME, DB_VERSION, context)
+        val reservationDAO = ReservationDAOImpl.getInstance(database)
+        val local = ReservationLocalDatasource.getInstance(reservationDAO)
+        return ReservationRepositoryImpl.getInstance(local)
+    }
+
+    fun getUserRepository(context: Context): UserRepository {
+        val database = DatabaseHelper.getInstance(DB_NAME, DB_VERSION, context)
+        val userDAO = UserDAOImpl.getInstance(database)
+        val pref = PreferencesHelperImpl.getInstance(context)
+        val local = UserLocalDatasource.getInstance(userDAO, pref)
+        return UserRepositoryImpl.getInstance(local)
+    }
+
+    fun getLocationRepository(context: Context): LocationRepository {
+        val database = DatabaseHelper.getInstance(DB_NAME, DB_VERSION, context)
+        val locationDAO = LocationFavouriteDAOImpl.getInstance(database)
+        val pref = PreferencesHelperImpl.getInstance(context)
+        val local = LocationLocalDatasource.getInstance(locationDAO, pref)
+        val remote = LocationRemoteDatasource.getInstance()
+        return LocationRepositoryImpl.getInstance(local, remote)
+    }
+
+    fun getHotelRepository(context: Context): HotelRepository {
+        val database = DatabaseHelper.getInstance(DB_NAME, DB_VERSION, context)
+        val locationDAO = LocationFavouriteDAOImpl.getInstance(database)
+        val pref = PreferencesHelperImpl.getInstance(context)
+        val local = HotelLocalDatasource.getInstance(locationDAO, pref)
+        val remote = HotelRemoteDatasource.getInstance()
+        return HotelRepositoryImpl.getInstance(local, remote)
+    }
+
+    fun getRestaurantRepository(context: Context): RestaurantRepository {
+        val database = DatabaseHelper.getInstance(DB_NAME, DB_VERSION, context)
```

Comment:
những cái thuộc tính DB_NAME, DB_VERSION thì chỉ cần để private trong DatabaseHelper là được 
tham số truyền vào chỉ cần Context
Thứ tự các paramters thì context luôn được ưu tiên đứng đầu 
```suggestion
        val database = DatabaseHelper.getInstance(context)
```